MekHQ/src/mekhq/campaign/mission/AtBContract.java

```diff
@@ -1908,17 +1921,30 @@ private static int getAverageBattleValue(Campaign campaign, String factionCode,
         int entries = unitTable.getNumEntries();
 
         int totalBattleValue = 0;
+        int totalGBV = 0;
         int rollingCount = 0;
 
         for (int i = 0; i < entries; i++) {
-            int weight = unitTable.getEntryWeight(i);
-            int battleValue = unitTable.generateUnit().getBV();
-
+            int battleValue = unitTable.getBV(i); // 0 for salvage
+            if (0 == battleValue) {
+                // Removing this check will break things, see the other comments.
+                continue;
+            }
+            // TODO implement getGBV(int index) in UnitTable to simplify this?
+            // getMekSummary(int index) is NULL for salvage.
+            int genericBattleValue = unitTable.getMekSummary(i).loadEntity().getGenericBattleValue();
+            int weight = unitTable.getEntryWeight(i); // NOT 0 for salvage
+            
             totalBattleValue += battleValue * weight;
+            totalGBV += genericBattleValue * weight;
             rollingCount += weight;
         }
 
-        return totalBattleValue / rollingCount;
+        if (campaign.getCampaignOptions().isUseGenericBattleValue()) {
+            return ((double) totalBattleValue) / totalGBV;
```

Comment:
Would you be able to explain why we're dividing BV2 by GBV here? I'm not suggesting it's wrong to do so, but I'm trying to get a sense of what returns we're expecting, so I can factor it into my future work.

Reply:
The basic assumption is that difficulty depends on the ratio of player (+allies) BV2 to enemy BV2.
Since FG3 matches GBV it matters less what BV2 the average Mek got, but rather how many points of BV2 it gets per point of GBV.

It's less extreme for the enemies and allies that should be using fairly balanced RATs, but a player achieving an average of ~2500 BV2 per unit entirely with Medium or lighter Clan Meks will face a much weaker OpFor than one achieving the same BV2 average with IS Assault Meks, because the latter gives the OpFor a much larger GBV budget to use.

Reply:
I think I can see the logic! :)

To confirm: when we communicate challenge to the player: if I’m displaying half a ‘skull’ per 20% variance between player forces and enemy that will still work with this approach? Or will I need to rethink?

As an illustration, we’re going to be shifting to display skulls as a way of illustrating difficulty. 2.5 skulls would be roughly balanced forces. 2 skulls would have the player force roughly exceed the enemy by about 20%, while 3 skulls would have the enemy power approximately 20% more powerful than the player. And so on, with each half-skull equating to roughly 20% variance.

Reply:
Generally yes.
For forces with similar weight distribution and therefore similar GBV, the output of the method shouldn't change other than what the bug fix did.
But I have not run the numbers, so I can't say for sure.

I think I understand what you're doing.
0 skulls = 0% of player BV2 (actually up to 10%)
2.5 skulls = 100% (+-10%)
5 skulls = 200%
That seems like a usable range.
Though with the combination of higher tech and higher skill levels, it's entirely possible for an OpFor to go far beyond 200%. I don't think the difference between 300% and 400% would matter, the player should lose either way, but you could use the reciprocal for the upper half of the scale so it covers all possible BV ratios.
As in 3 skulls mean the player's BV2 is 20% lower than the enemy's and so on, with 4.5 skulls meaning around 500% enemy BV (the player having 80% less) and 5 skulls being the same edge case as 0 skulls, one side having 10 or more times the average BV of the other.

That's a design choice though, more granularity in the range that should actually matter vs being able to differentiate between two extremely bad ideas.

Reply:
Perfect, thank you